### DISCUSSION TITLE
should ACE_Process_Options::setenv allow adding a variable that is already in it's environment_buffer?

### DISCUSSION THREAD

**Discussion by yougotborked** [DOCGroup/ACE_TAO#2099] [Q&A]
Especially on Windows when using inheritEnvironment to pre-populate the buffer, once we pass it along to spawn() ,
I have found my additional setenv call does not end up replacing the first one.

I feel like setenv should modify any variables that are already in it's buffer. not just keep appending to the buffer.

I'm happy to provide a PR if the community/maintainers think this is worth changing.

**Comment by mitza-oci** [comment]
I think having it match POSIX `setenv` behavior as much as possible would be good.
